### DISCUSSION TITLE
Text-to-speech (TTS) functionality

### DISCUSSION THREAD

**Discussion by ejuarezg** [ahrm/sioyek#485] [Ideas]
Hi everyone,

I'm really enjoying using Sioyek and was wondering if anyone else was interested in text-to-speech (TTS) functionality. I think it would be really cool to have the ability to play back PDFs. This post is to gauge your interest and talk about important considerations of adding TTS support to Sioyek. If it's feasible, I may start working on this functionality as a side project (collaborators are welcomed). 

So far, I only have only thought about the following things

- Using a python extension (something like https://github.com/ahrm/sioyek-python-extensions) to copy the text selected in a PDF
- Feeding the text to a TTS solution like https://github.com/mozilla/TTS (I would need help with this)

**Comment by ahrm** [comment]
I did exactly that a while back and wrote a blog post about it:
https://ahrm.github.io/jekyll/update/2022/07/05/implementing-a-screen-reader-for-sioyek.html

It was before the current extension api so some of the steps are a little different now. The reason I didn't include it as part of sioyek extensions is that it requires a lot of dependencies which are pretty heavy(TTS and aeneas which is pretty hard to install on windows).

**Reply by ejuarezg** [reply]
I'll have to take a look at this again, thanks for reminding me. I would love to hack on this and see how much further we can simplify the process.

**Reply by temberature** [reply]
I implement TTS for PDF according to the blog post last several days, I write a simple note for it.
https://github.com/temberature/blog/issues/330

Thanks for the great job.
